controllers/volumereplicationgroup_controller.go

```diff
@@ -898,15 +898,98 @@ func (v *VRGInstance) createVR(vrNamespacedName types.NamespacedName, state volr
 			Namespace: vrNamespacedName.Namespace,
 		},
 		Spec: volrep.VolumeReplicationSpec{
-			DataSource: &corev1.TypedLocalObjectReference{
+			DataSource: corev1.TypedLocalObjectReference{
 				Kind: "PersistentVolumeClaim",
 				Name: vrNamespacedName.Name,
 			},
-			State: state,
+			ReplicationState: state,
+
+			VolumeReplicationClass: v.instance.Spec.VolumeReplicationClass,
 		},
 	}
 
-	return v.reconciler.Create(v.ctx, cr)
+	// Let VRG receive notification for any changes to VolumeReplication CR
+	// created by VRG.
+	if err := ctrl.SetControllerReference(v.instance, cr, v.reconciler.Scheme); err != nil {
+		return fmt.Errorf("failed to set owner reference to VolumeReplication resource (%s/%s)",
+			cr.Name, cr.Namespace)
+	}
+
+	v.log.Info("Creating VolumeReplication resource", "resource", cr)
+
+	if err := v.reconciler.Create(v.ctx, cr); err != nil {
+		return fmt.Errorf("failed to create VolumeReplication resource (%s)", vrNamespacedName)
+	}
```

Comment:
(nit) This comment is not about this PR.  It may help to use the variable name "volRep" instead of "cr" for consistency with other methods.

Reply:
ACK.

If I will have to make any additional changes to this PR (to address some other review comment), I will make this change as well.